### PR TITLE
4BudGet RoR Capstone project

### PR DESCRIPTION
**Pull request by PabloBona**
### Implemented Features:
1. **Splash Screen:**
   - Created a simple splash screen with the app name and links to sign up and log in pages.

2. **Sign Up and Log In Pages:**
   - Implemented user registration with mandatory fields: full name, email, and password.
   - Enabled user login using email and password credentials.
   - Implemented access control: users must be logged in to access specific pages.

3. **Home Page (Categories Page):**
   - Implemented a categories page where users can view category names, icons, and total transaction amounts.
   - Implemented navigation to transaction pages for each category.
   - Added a button for users to add new categories.

4. **Transactions Page:**
   - Displayed transactions for a selected category, sorted by most recent.
   - Showed the total amount for the selected category.
   - Implemented navigation to add new transactions page.
   - Enabled navigation back to the home page.

5. **"Add a New Category" Page:**
   - Created a form for users to add new categories with mandatory fields: name and icon.
   - Implemented functionality to save new categories and navigate back to the home page.
   - Enabled navigation back to the home page.

6. **"Add a New Transaction" Page:**
   - Implemented a form for users to add new transactions with mandatory fields: name, amount, and at least one category.
   - Implemented functionality to save new transactions and navigate back to the transactions page for that category.
   - Enabled navigation back to the transactions page for that category.

### Technical Details:
- **Database:** Utilized PostgreSQL for data storage.
- **Authentication:** Implemented user authentication using Devise.
- **Input Validation:** Validated all user inputs to prevent malicious actions and ensure data integrity.
- **Testing:** Created unit and integration tests for all important components of the Rails application.

### Deployment:
The project has been deployed and is accessible online. You can find the deployed application at [4BudGet](https://budget-app-y2fh.onrender.com)].

I kindly request you to review my changes and merge this feature branch into the `develop` branch. I am confident that the implemented features meet the project requirements and have been thoroughly tested to ensure their functionality and security.

Thank you for your time and consideration.

Best regards,

### CODE REVIEW

**Review by TedLivist — COMMENTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @PabloBona,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly set up your linters correctly and ensure it is running on the PR before submitting for reviews.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by PabloBona** [comment]
Thank you for letting me know @TedLivist . I appreciate your feedback. I've corrected the issues with the linters, and I have ensured that they are running correctly on the pull request now.

Please let me know if there's anything else I can do to assist you further.
image

**Review by julie-ify — COMMENTED** [review]
Hi @PabloBona,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* Your linters may be passing locally, but it is a requirement that you should set up Github Actions to run on this pull request. Just as you likely have done in previous projects and modules, the commits should have a green tick next to them on the pull request. You may need to visit [the Microverse ROR linters configuration](https://github.com/microverseinc/linters-config/tree/master/ror) and set up the `.github/workflows` directory

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by PabloBona** [comment]
@julie-ify 
I'm sorry! It's entirely my fault!
Thank you for your patience.

File: app/views/payments/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<%= render "shared/header", title: 'NEW PAYMENT', icon: 'arrow-back', link: root_path, logout: true %>
```
Comment (julie-ify) [inline app/views/payments/new.html.erb]: - [ ] When I am on the new transaction page and I click the back arrow I should be taken to the transactions index page, not the root path. Kindly fix that.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,163 @@
-# 4budget
-Our budget planner and daily budget tracker is a money saving app that makes managing your personal finances simple!
+# 4Budget: Personal Finance Management App
+
+## 📗 Table of Contents
+
```
Comment (julie-ify) [inline README.md:4]: - [ ] Your readme file should contain an Acknowledgement section that gives credit to Gregoire Vella the app designer. Kindly add it.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @PabloBona,

Good job so far!

## Highlights!
- Good app deployment ✔️ 
- Good app design and UI ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (PabloBona) [reply README.md:4]: "Thank you for your feedback, @julie-ify ! I just finished it and I'm asking for another review! Greetings!"

File: README.md [hunk 3]
```diff
@@ -47,6 +48,11 @@ This project is inspired by the original design idea by Gregoire Vella on Behanc
 ![Alt text](image-1.png)
 ![Alt text](image-2.png)
 ![Alt text](image-3.png)
+
+### Acknowledgment <a name="Acknowledgment"></a>
+
+We would like to express our deepest gratitude to **Gregoire Vella**, the talented app designer who contributed significantly to the visual and user experience aspects of this project. His creative insights and dedication played a crucial role in shaping the design of our application. We are truly thankful for his expertise and valuable contributions.
+
```
Comment (Shedrack-Sunday) [inline README.md]: - [ ] Firstly, kindly add the associated links to the name of the design owner. This way, potential people of interest in your work will have a path to navigate, to confirm the authenticity of the owner 👍  

- [ ] Secondly, the `Acknowledgement` section of the README is positioned at the bottom of the README setup, as this particular detail is meant to be at the concluding part of your work to give credit.
- Please adjust the position as in the [Template](https://github.com/microverseinc/readme-template) 👍 

### Template:
image

### Design owner needed links:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @PabloBona        👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the Rails features ✔️
- Kudus fixing the transaction path as specified by the last reviewer 👏 

# - Required Changes  🌵  

- [ ] Kindly ensure all detailed information regarding the project requirements and templates given are inputted properly 

_Check the comment below

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by addod19 — APPROVED** [review]
Hi @PabloBona, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Nice job fixing the README file and making it more professional :rocket: 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**